components/world-4/breeding/pets.tsx

```diff
@@ -0,0 +1,532 @@
+                />
+                <SelectMultiple
+                    size="small"
+                    placeholder="Filter by"
+                    searchPlaceholder="Search bonuses"
+                    value={filter}
+                    options={currentFilterOptions}
+                    onChange={({ value: nextValue }) => { setFilter(nextValue); }}
+                    onSearch={text => {
+                        // The line below escapes regular expression special characters:
+                        // [ \ ^ $ . | ? * + ( )
+                        const escapedText = text.replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&');
+
+                        // Create the regular expression with modified value which
+                        // handles escaping special characters. Without escaping special
+                        // characters, errors will appear in the console
+                        const exp = new RegExp(escapedText, 'i');
+                        setCurrentFilterOptions(allFilterOptions.filter(o => exp.test(o)));
+                    }}
+                    onClose={() => setCurrentFilterOptions(allFilterOptions)}
+                />
+            </Box>
+            <TextAndLabel label='Shiny speed multiplier :' text={`${round(breeding.getShinySpeed(starSignEquipped, silkRodeChip))}x`}/>
+            <Grid columns={size == "small" ? ["1"] : ["1/3", "1/3", "1/3"]} fill>
+                {
+                    petsToShow?.filter(pet => pet.data.petId != "_").map((pet, pIndex) => {
+                        const petInFenceyard: number = breeding.fenceyardPets.filter(fenceyardPet => fenceyardPet.data.petId == pet.data.petId && fenceyardPet.gene.index == 5).length;
+                        const enemy = EnemyInfo.find(enemy => enemy.id == pet.data.petId);
+                        const shinySpeed = breeding.getShinySpeed(starSignEquipped, silkRodeChip);
+                        const timeToNextLevel = (pet.getNextShinyGoal() - pet.shinyProgress) * 84600 / shinySpeed;
+
+                        return (
+                            petInFenceyard > 0 ?
+                                <ShadowBox background="dark-1" border={{ size: '2px', style: 'solid', color: 'green' }} key={pIndex} direction="row" gap="medium" margin={{ bottom: 'medium', right: 'small' }} align="center" pad="small" style={{ opacity: pet.shinyLevel > 0 ? 1 : .5 }}>
+                                    <IconImage data={{ location: enemy?.id.toLowerCase() ?? "Unknown", width: 67, height: 67 }} style={{ paddingBottom: '15px' }} />
+                                    <Grid columns={["45%", "45%", "10%"]} fill align="center">
+                                        <Box>
+                                            <Text size="16px">Lvl: {pet.shinyLevel}</Text>
+                                            <Text size="16px">{nFormatter(Math.floor(pet.shinyProgress))}/{nFormatter(pet.getNextShinyGoal())} days</Text>
```

Comment:
Happy to hear your thoughts on this but what do you think about swapping this information with the "Time to next level" information?

I think `35/230 days` isn't really information I care about but hovering and seeing "82 days till next level" is really cool to have at a glance.

As a whole I think the tooltip is actually what has the most interesting info and it's not easily visible so I think that's a missed opportunity.

Again happy to hear what you think but I'd be keen to see a world where we shuffle information on this page to make the tooltip info easily available at a glance.

Reply:
![image](https://github.com/user-attachments/assets/cadc4f09-095d-42f5-8652-b87afe877839)

I made a few changes, and now it looks like that for pets that are currently in fence yard, the display won't change for other pets as it's useless to give time to next level when there's no pets available